projects/plugins/jetpack/_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php

```diff
@@ -834,36 +870,41 @@ public function update_data( $request ) {
 				case 'do_not_track':
 				case 'version':
 				case 'collapse_nudges':
-					$grouped_options          = $grouped_options_current = (array) get_option( 'stats_options' );
-					$grouped_options[$option] = $value;
+					$grouped_options_current    = (array) get_option( 'stats_options' );
+					$grouped_options            = $grouped_options_current;
+					$grouped_options[ $option ] = $value;
 
 					// If option value was the same, consider it done.
-					$updated = $grouped_options_current != $grouped_options ? update_option( 'stats_options', $grouped_options ) : true;
+					$updated = $grouped_options_current !== $grouped_options
+						? update_option( 'stats_options', $grouped_options )
+						: true;
 					break;
 
 				case 'akismet_show_user_comments_approved':
-
-					// Save Akismet option '1' or '0' like it's done in akismet/class.akismet-admin.php
-					$updated = get_option( $option ) != $value ? update_option( $option, (bool) $value ? '1' : '0' ) : true;
+					// Save Akismet option '1' or '0' like it's done in akismet/class.akismet-admin.php.
+					$updated = get_option( $option ) != $value // phpcs:ignore Universal.Operators.StrictComparisons.LooseNotEqual
+						? update_option( $option, (bool) $value ? '1' : '0' )
```

Comment:
Since we're touching this anyway, I note the `(bool)` cast is redundant.
```suggestion
						? update_option( $option, $value ? '1' : '0' )
```
I also had to stop and verify that `(bool) $value ? '1' : '0'` works like `( (bool) $value ) ? '1' : '0'` rather than `(bool) ( $value ? '1' : '0' )`.